src/hooks/useApr.ts

```diff
@@ -0,0 +1,101 @@
+import { useStore } from 'src/store';
+import { computed, ref, watchEffect } from 'vue';
+import { useChainMetadata, useCurrentEra, useTvl } from '.';
+import { defaultAmountWithDecimals, reduceBalanceToDenom } from './helper/plasmUtils';
+import { useApi } from './useApi';
+
+// Ref: https://github.com/PlasmNetwork/Astar/blob/5b01ef3c2ca608126601c1bd04270ed08ece69c4/runtime/shiden/src/lib.rs#L435
+// Memo: 50% of block rewards goes to dappsStaking, 50% goes to block validator
+// Fixme: ideally get the value from API
+const DAPPS_REWARD_RATE = 0.5;
+
+const TS_FIRST_BLOCK = {
```

Comment:
instead of hard-coding all block timestamps for each blocks, I recommend using the `await api.rpc.chain.getBlockHash(1);` and `await api.query.at(blockHash).timestamp.now();` to get the past block timestamp.

Reply:
yes, I was looking for something like this for a very long time. I think `await api.query.at(blockHash).timestamp.now()` is not callable, instead it should be queried like [this](https://github.com/polkadot-js/api/issues/3970#issuecomment-922492767). However, it doesn't work for our (Shiden) API somehow...

hash: 0x985b77fdbc66f3ea8316b7107b86e8542c70dd045eed26f123795903d8e9cba3
 
<img width="954" alt="image" src="https://user-images.githubusercontent.com/92044428/143478236-decf4631-8467-4a24-a85d-a665710123b7.png">

![image](https://user-images.githubusercontent.com/92044428/143478549-59d5179a-8daa-4aed-8054-c8add7011ed3.png)

![image](https://user-images.githubusercontent.com/92044428/143478566-53fc9392-1fb8-43f9-b879-99831919e788.png)

Reply:
This seems like a storage MetaData V4 error that we were seeing before.
![Screen Shot 2021-11-25 at 5 58 53 PM](https://user-images.githubusercontent.com/40356749/143479988-6c7e243b-e0bb-4011-9f12-ca66c0782e3b.png)
I thought this was solved in https://github.com/PlasmNetwork/astar-apps/pull/167, but maybe not?
@bobo-kovacevic do you know the cause of this?

Reply:
@hoonsubin 
How about improving the timestamp part once MetaData issue has been fixed? Then perhaps we can merge this APR feature first?

Reply:
yeah, that sounds good